src/main/java/core/basesyntax/Main.java

```diff
@@ -0,0 +1,18 @@
+package core.basesyntax;
+
+public class Main {
+    private static final int ARRAY_COUNT = 6;
+
+    public static void main(String[] args) {
+        FigureSupplier supplier = new FigureSupplier();
+        Figure[] figure = new Figure[ARRAY_COUNT];
+        for (int i = 0; i < ARRAY_COUNT; i++) {
+            if (i < ARRAY_COUNT / 2) {
```

Comment:
better to not rely on the constant but on the actual instance
```suggestion
        for (int i = 0; i < figures.length; i++) {
            if (i < figures.length / 2) {
```